### ISSUE TITLE
"indent" rule false positive in "switch" statement

### ISSUE THREAD

**Issue by feross** [eslint/eslint#3001]
Using the code on `master` (as of https://github.com/eslint/eslint/commit/1c70834295b064021e6aa6698473375063bffbae), I get a false positive with the `indent` rule enabled.

```js
function salutation () {
  switch (1) {
    case 0: return console.log('hi')
    case 1: return console.log('hey')
  }
}
```

Rule:

```
"indent": [2, 2]
```

Errors:

```
t.js:3:13: Expected indentation of 6 characters.
t.js:4:13: Expected indentation of 6 characters.
```

**Comment by gyandeeps** [comment]
what was the config used for the rule?

**Comment by feross** [comment]
```
"indent": [2, 2]
```

**Comment by gyandeeps** [comment]
This rule is reporting incorrectly because you dont have `;` at the end of returns.
I will further investigate whats going on here.

**Comment by gyandeeps** [comment]
I think this maybe an issue with Espree.
Switch case node has an line start of 3 and line end of 4. which is incorrect.

But I checked [esprima](http://esprima.org/demo/parse.html?code=function%20test()%7B%0Aswitch(a)%20%7B%0A%20%20%20%20case%20%22a%22%3A%20return%20console.log(%22A%22)%0A%20%20%20%20case%20%22b%22%3A%20return%20console.log(%22B%22)%0A%7D%0A%7D), it parses it correctly.

**Comment by gyandeeps** [comment]
Issue logged to Espree: https://github.com/eslint/espree/issues/164

**Comment by lo1tuma** [comment]
working on this

**Comment by mistadikay** [comment]
Still can reproduce this bug on 1.0.0-rc-1

**Comment by ilyavolodin** [comment]
Hmmm... @lo1tuma added unittests to verify reported scenario. If yours differ, can you open a new issue?

**Comment by gyandeeps** [comment]
@mistadikay was missing the indent case option.
https://github.com/eslint/eslint/issues/3024
